packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,250 @@
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+
+import "./GachaLazyClaim.sol";
+import "./IERC1155GachaLazyClaim.sol";
+
+/**
+ * @title Gacha Lazy 1155 Payable Claim
+ * @author manifold.xyz
+ * @notice
+ */
+contract ERC1155GachaLazyClaim is IERC165, IERC1155GachaLazyClaim, ICreatorExtensionTokenURI, GachaLazyClaim {
+  using Strings for uint256;
+
+  // stores mapping from contractAddress/instanceId to the claim it represents
+  // { contractAddress => { instanceId => Claim } }
+  mapping(address => mapping(uint256 => Claim)) private _claims;
+
+  // { contractAddress => { tokenId => { instanceId } }
+  mapping(address => mapping(uint256 => uint256)) private _tokenInstances;
+
+  function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, AdminControl) returns (bool) {
+    return
+      interfaceId == type(IERC1155GachaLazyClaim).interfaceId ||
+      interfaceId == type(IGachaLazyClaim).interfaceId ||
+      interfaceId == type(ICreatorExtensionTokenURI).interfaceId ||
+      interfaceId == type(IAdminControl).interfaceId ||
+      interfaceId == type(IERC165).interfaceId;
+  }
+
+  constructor(address initialOwner) GachaLazyClaim(initialOwner) {}
+
+  /**
+   * See {IERC1155GachaLazyClaim-initializeClaim}.
+   */
+  function initializeClaim(
+    address creatorContractAddress,
+    uint256 instanceId,
+    ClaimParameters calldata claimParameters
+  ) external payable override adminRequired {
+    if (instanceId == 0 || instanceId > MAX_UINT_56) revert IGachaLazyClaim.InvalidInstance();
+
+    // Checks
+    if (claimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    if (claimParameters.endDate != 0 && claimParameters.startDate >= claimParameters.endDate) revert IGachaLazyClaim.InvalidDate();
+    
+    address[] memory receivers = new address[](1);
+    receivers[0] = msg.sender;
+    uint256[] memory amounts = new uint256[](claimParameters.itemVariations);
+    string[] memory uris = new string[](claimParameters.itemVariations);
+    uint256[] memory newTokenIds = IERC1155CreatorCore(creatorContractAddress).mintExtensionNew(receivers, amounts, uris);
+
+    require(newTokenIds[0] <= MAX_UINT_80, "Token ID exceeds uint80 range");
+
+    // Create the claim
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claimParameters.storageProtocol,
+      total: 0,
+      totalMax: claimParameters.totalMax,
+      startDate: claimParameters.startDate,
+      endDate: claimParameters.endDate,
+      startingTokenId: uint80(newTokenIds[0]),
+      itemVariations: claimParameters.itemVariations,
+      location: claimParameters.location,
+      paymentReceiver: claimParameters.paymentReceiver,
+      cost: claimParameters.cost,
+      erc20: claimParameters.erc20
+    });
+    for (uint8 i; i < claimParameters.itemVariations; i++) {
```

Comment:
use uint256 for incrementation, it's more gas efficient.  Using anything less than uint256 results in a cost to cast (uint256 is the native machine language construct)

Reply:
To save gas, you can do an:

unchecked { i++; } within the for loop and get rid of i++.  See other contracts for example